### PR TITLE
0.0.6 - allow client to store encrypted privateKey in vault

### PR DESCRIPTION
**Pull request by mrose17**
gated on https://github.com/brave/vault-client/pull/7

### CODE REVIEW

File: documentation/Principles.md [hunk 1]
```diff
@@ -83,25 +90,40 @@ The `signature` and `nonce` properties are used to ensure that the client actual
...
+          var iv = window.crypto.getRandomValues(new Uint8Array(12))
+          var message = { header           : {}
+                        , payload          : 
+                          { version        : 1
+                          , publicKey      : '04' +
+                                              new Buffer(publicKey.x, 'base64').toString('hex') +
+                                              new Buffer(publicKey.y, 'base64').toString('hex')
+                          , privateKey      : 
+                            { encryptedData : ab2hex(ciphertext)
+                            , iv            : ab2tex(iv)
```
Comment (diracdeltas) [inline documentation/Principles.md]: ab2hex, not ab2tex

File: src/controllers/user.js [hunk 2]
```diff
@@ -86,15 +81,17 @@ v1.put =
     user = await users.findOne({ userId: userId })
     createP = !user
     if (createP) {
-      // NB: payload.version should be mandatory
-      if ((payload.version) && (payload.version !== 1)) {
+      if ((!payload.version) && (payload.version !== 1)) {
```
Comment (diracdeltas) [inline src/controllers/user.js]: shouldn't this be `if (!payload.version || payload.version !==1)`? can be simplified to just `if (payload.version !==1)`.

Comment (mrose17) [inline src/controllers/user.js]: right!

File: documentation/Principles.md (hunk 1 above)
Comment (mrose17) [inline documentation/Principles.md]: d'oh

**Comment by mrose17** [comment]
fix #49

**Pull request merged**